os/src/Path.scala

```diff
@@ -484,6 +484,7 @@ class Path private[os] (val wrapped: java.nio.file.Path)
     new SeekableSource.ChannelSource(java.nio.file.Files.newByteChannel(wrapped))
 
   require(wrapped.isAbsolute || Path.driveRelative(wrapped), s"$wrapped is not an absolute path")
+  def root = Option(wrapped.getRoot).map(_.toString).getOrElse("")
```

Comment:
Does root have to be a string? I wonder if we can make it a `java.nio.file.Path`, or some opaque wrapper of one `trait os.RootPath{ def toString: String }` or something. 

We could keep the `def root(String)` factory method for convenience, but making the `def root` somewhat less concrete would give us flexibility in future if we find we want to expose more of the underlying `java.nio` capabilities

Reply:
I am not sure. If we knew how it could be used, we could leave that open, but putting just the String there seems to be the simplest way to go. The Path needs additional checks, and I am not sure what it would mean - most of the Paths are not just roots, and even if they are, one is able to construct os.Path from them in another way than os.root. Unless you mean extracting just the root from the nio Path?

Reply:
One case that was mentioned in the other thread: `java.nio.file.Path`s are bound to a specific `java.nio.file.Filesystem`. And can refer to e.g. in-memory filesystems or zip-filesystems and other things. 


```scala
lihaoyi mill$ amm
Loading...
Welcome to the Ammonite Repl 2.5.4 (Scala 2.13.8 Java 17.0.6)
@ import collection.JavaConverters._
import collection.JavaConverters._

@ val zipFs = java.nio.file.FileSystems.newFileSystem(java.net.URI.create("jar:file:///usr/local/bin/amm"), Map("create" -> "true").asJava)
zipFs: java.nio.file.FileSystem = /usr/local/bin/amm

@ java.nio.file.Paths.get("/") == java.nio.file.Paths.get("/")
res2: Boolean = true

@ zipFs.getPath("/") == java.nio.file.Paths.get("/")
res3: Boolean = false

@ zipFs.getPath("/").getFileSystem
res4: java.nio.file.FileSystem = /usr/local/bin/amm

@ java.nio.file.Paths.get("/").getFileSystem
res5: java.nio.file.FileSystem = sun.nio.fs.MacOSXFileSystem@4228bf58
```

This raises two questions:

1. How does that work on windows? Is each windows "drive" a separate `Filesystem`, or are they all on one filesystem with a different `getRoot` path?

2. Should `os.root` be able to refer to different filesystems? Given that we are right now considering how to flesh out `os.root` to make it less anemic, I feel like we might as well flesh it out "all the way". Otherwise we may end up coming back and having to repeat this whole process again later.

I feel like having `os.Path#root` return a non-string seems like the right thing to do; `java.nio.file.Path` is more complex than just a string, and we should model that somehow. Having it return `trait Root{ def prefix: String; def filesystem: java.nio.file.Filesystem }` seems like a more future-proof way of specifying it

Reply:
Those are fair points; I will investigate.

Reply:
Sounds good to me.